index.js

```diff
@@ -1,6 +1,23 @@
 const core = require("@actions/core");
 const puppeteer = require("puppeteer-core");
 const io = require("@actions/io");
+const os = require("os");
+
+function getChromePath() {
+  let browserPath = "";
+
+  if (os.type() === "Windows_NT") {
+    // Chrome is usually installed as a 32-bit application, on 64-bit systems it will have a different installation path.
+    const programFiles = os.arch() === 'x64' ? process.env["PROGRAMFILES(X86)"] : process.env.PROGRAMFILES;
+    browserPath = path.join(programFiles, "Google/Chrome/Application/chrome.exe");
+  } else if (os.type() === "Darwin") {
+    browserPath = "/Application/Google Chrome.app/Contents/MacOS/Google Chrome";
```

Comment:
Just a thought, this would be great if the MacOS runner had chrome, but this could confuse people as puppeteer would throw an error saying it couldn't find a binary. Maybe instead you could throw an error here and fail the task on MacOS so we could more explicitly say MacOS is not supported for the time being?

Reply:
I initially thought just letting it fail would suffice.
But I agree, I think it is better to be explicit.